flake.nix

```diff
@@ -0,0 +1,21 @@
+{
+  description = "FormalSystems lean implementation";
+
+  inputs.lean.url = "github:leanprover/lean4";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+
+  outputs = { self, lean, flake-utils }: flake-utils.lib.eachDefaultSystem (system:
+    let
+      leanPkgs = lean.packages.${system};
+      pkg = leanPkgs.buildLeanPackage {
+        name = "FormalSystems";
+        src = ./.;
+      };
+    in {
+      packages = pkg // {
+        inherit (leanPkgs) lean;
+      };
```

Comment:
This is probably not what you want, you are basically doing something like this:
```nix
packages = leanPkgs.buildLeanPackage {
  name = "FormalSystems";
  src = ./.;
  lean = leanPkgs.lean;
};
```

I guess what you want is rather this:
```nix
packages = {
  inherit pkg;
  inherit (leanPkgs) lean;
};
```